tests/population/test_mortality.py

```diff
@@ -33,6 +51,169 @@ def test_mortality_default_lookup_configuration(
+    assert pop1.loc[dead_idx, "exit_time"].unique()[0] == sim._clock._clock_time
+
+
+def test_mortality_cause_of_death(
+    fuzzy_checker: FuzzyChecker,
+    base_config,
+    full_simulants,
+    base_plugins,
+    generate_population_mock,
+):
+
+    start_population_size = len(full_simulants)
+    generate_population_mock.return_value = full_simulants.drop(columns=["tracked"])
+    bp = BasePopulation()
+    mortality = Mortality()
+    # Set up Disease model
+    year_start = base_config.time.start.year
+    year_end = base_config.time.end.year
+
+    healthy = BaseDiseaseState("healthy")
+    mort_get_data_funcs = {
+        "dwell_time": lambda _, __: pd.Timedelta(days=0),
+        "disability_weight": lambda _, __: 0.0,
+        "prevalence": lambda _, __: build_table(
+            0.5, year_start - 1, year_end, ["age", "year", "sex", "value"]
+        ),
+        "excess_mortality_rate": lambda _, __: build_table(0.7, year_start - 1, year_end),
+    }
+    mortality_state = DiseaseState("sick", get_data_functions=mort_get_data_funcs)
+    healthy.add_transition(Transition(healthy, mortality_state))
+
+    model = DiseaseModel("test", initial_state=healthy, states=[healthy, mortality_state])
+    sim = InteractiveContext(
+        components=[bp, mortality, model], plugin_configuration=base_plugins, setup=False
+    )
+    override_config = {
+        "population": {
+            "population_size": start_population_size,
+            "include_sex": "Male",
+        },
+        "mortality": {
+            "all_cause_mortality_rate": {
+                "value": 0.8,
+            }
+        },
+    }
+    sim.configuration.update(override_config)
+    sim.setup()
+    sim.step()
+    # Only other causes and sick for cause of death
+    pop1 = sim.get_population()
+    for cause_of_death in ["other_causes", "sick"]:
+        dead = pop1.loc[pop1["cause_of_death"] == cause_of_death]
+        # Disease model seems to set mortality rate for that diesease back to 0
+        # if a simulant dies from it
+        rates = mortality.mortality_rate(pop1.index)[cause_of_death].unique()
+        for mortality_rate in rates:
+            if mortality_rate == 0:
+                continue
+            else:
+                rate = mortality_rate
+            if cause_of_death == "sick":
+                rate *= 0.5  # prevalence
+            fuzzy_checker.fuzzy_assert_proportion(
+                name=f"test_mortality_rate_{cause_of_death}",
+                observed_numerator=len(dead),
+                observed_denominator=len(pop1),
+                target_proportion=rate,
+            )
+
+
+def test_mortality_ylls(setup_sim_with_pop_and_mortality):
+    sim, bp, mortality = setup_sim_with_pop_and_mortality
+    sim.step()
+    pop1 = sim.get_population()
+
+    dead_idx = pop1.index[pop1["alive"] == "dead"]
+    ylls = pop1.loc[dead_idx, "years_of_life_lost"]
+    assert (ylls == mortality.lookup_tables["life_expectancy"](dead_idx)).all()
```

Comment:
Note: this doesn't test that we correctly populated the life expectancy lookup table.